diff -up libbonobo-2.24.2/activation-server/object-directory-load.c.dont-tank-on-invalid-server-files libbonobo-2.24.2/activation-server/object-directory-load.c
--- libbonobo-2.24.2/activation-server/object-directory-load.c.dont-tank-on-invalid-server-files	2012-03-01 15:00:02.614782740 -0500
+++ libbonobo-2.24.2/activation-server/object-directory-load.c	2012-03-01 15:02:05.347778961 -0500
@@ -441,7 +441,7 @@ od_start_element (GMarkupParseContext *c
                 break;
                 break;
         default:
-                g_error ("start element, unknown state: %d", info->state);
+                g_printerr ("start element, unknown state: %d", info->state);
         }
 }
 
@@ -537,7 +537,7 @@ od_end_element (GMarkupParseContext *con
         case STATE_START:
                 break;
         default:
-                g_error ("end element, unknown state: %d", info->state);
+                g_printerr ("end element, unknown state: %d", info->state);
         }
 }
 
@@ -548,9 +548,9 @@ od_error (GMarkupParseContext *context,
 {
         ParseInfo *info = user_data;
 
-        g_error ("Failed to parse: '%s' in file '%s'",
-                 error ? error->message : "<nomsg>",
-                 info->filename ? info->filename : "<memory>");
+        g_printerr ("Failed to parse: '%s' in file '%s'",
+                    error ? error->message : "<nomsg>",
+                    info->filename ? info->filename : "<memory>");
 }
 
 static GMarkupParser od_gmarkup_parser = {
